kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h

```diff
@@ -1055,14 +1086,26 @@ class ResidualBasedBlockBuilderAndSolver
                 if (constraint_is_active) {
                     it_const->CalculateLocalSystem(transformation_matrix, constant_vector, r_current_process_info);
                     it_const->EquationIdVector(slave_equation_ids, master_equation_ids, r_current_process_info);
+                    it_const->GetDofList(slave_dof_list, master_dof_list, r_current_process_info);
+
+                    // Set to zero the non fixed and solvable dofs
+                    IndexType column = 0;
+                    for (auto& master_dof : master_dof_list) {
+                        if (!(master_dof->IsFixed() || BaseType::mDofSet.find(*master_dof) != BaseType::mDofSet.end())) {
+                            for (IndexType i_zero = 0; i_zero < transformation_matrix.size1(); ++i_zero) {
+                                transformation_matrix(i_zero, column) = 0.0;

```

Comment:
matrix T has zero everywhere. It is set to zero on line 1056, so this line is in principle doing nothing

Reply:
If you comment i it fails...

Reply:
@loumalouomega i don't see this as a convincing argument

Reply:
you don't want to introduce any algebraic contribution in a de facto not active constraint

Reply:
currently the block builder and solver deliberately constructs the graph also for inactive elements and conditions. this is so to guarantee that when the elements/conditions are activated it is not needed to reconstruct the graph of the matrix.

I would say that we need to preserve this behaviour also for inactive Master/Slave constraints.

Having said this, i think that the mechanism i have implemented is not robust wrt, particularly the way i detect slaves during activation/deactivation. I am pretty sure that the origin of the error is in this.

Reply:
Is inactive "de facto". this means when the dof is not solved or fixed. It is not related with the flag ACTIVE